js/gulpfile.js

```diff
@@ -185,44 +185,125 @@ gulp.task("test:common:generate", (cb) => {
     pump([gulp.src(["../scripts/Controller.ice"]), slice2js(), gulp.dest("test/Common")], cb);
 });
 
+// A rollup plug-in to resolve "net" module as a mockup file.
+
+const nodeMokups = ["net", "fs", "path"];
+function NodeMookupResolver() {
+    return {
+        name: "mookup-resolver",
+        async resolveId(source) {
+            if (nodeMokups.includes(source)) {
+                return path.join(__dirname, `test/Common/${source}.mockup.js`);
+            }
+            return null;
+        },
+    };
+}
+
+gulp.task("ice:bundle", async (cb) => {
+    let bundle = await rollup({
+        input: "src/index.js",
+        plugins: [NodeMookupResolver()],
+    });
+    return bundle.write({
+        file: "dist/ice.js",
+        format: "esm",
+    });
+});
+
+// A rollup resolver plugin to resolve "ice" module as an external file.
+function IceResolver() {
+    return {
+        name: "ice-resolver",
+        async resolveId(source) {
+            if (source == "ice") {
+                return { id: "/ice.js", external: "absolute" };
+            }
+            return null;
+        },
+    };
+}
+
+gulp.task("test:common:bundle", async (cb) => {
+    let bundle = await rollup({
+        input: "test/Common/ControllerI.js",
+        plugins: [IceResolver()],
+    });
+    return bundle.write({
+        file: "dist/test/Common/ControllerI.js",
+        format: "esm",
+    });
+});
+
 gulp.task("test:common:clean", (cb) => {
-    del(["test/Common/Controller.js", "test/Common/.depend"]);
+    deleteAsync(["test/Common/Controller.js", "test/Common/.depend"]);
     cb();
 });
 
 const testTask = (name) => name.replace(/\//g, "_");
 const testCleanTask = (name) => testTask(name) + ":clean";
 const testBuildTask = (name) => testTask(name) + ":build";
+const testBundleTask = (name) => testTask(name) + ":bundle";
+const testAssetsTask = (name) => testTask(name) + ":assets";
 
 for (const name of tests) {
     gulp.task(testBuildTask(name), (cb) => {
-        const outdir = path.join(root, name);
+        const outputDirectory = path.join(root, name);
         pump(
             [
-                gulp.src(path.join(outdir, "*.ice")),
+                gulp.src(path.join(outputDirectory, "*.ice")),
                 slice2js({
-                    include: [outdir],
+                    include: [outputDirectory],
                 }),
-                gulp.dest(outdir),
+                gulp.dest(outputDirectory),
             ],
             cb,
         );
     });
 
-    gulp.task(testCleanTask(name), (cb) => {
+    gulp.task(testBundleTask(name), async (cb) => {
+        let input = fs.existsSync(path.join(name, "index.js"))
+            ? path.join(name, "index.js")
+            : path.join(name, "Client.js");
```

Comment:
We create a bundle for each test that is dynamically imported with `import()` the entry point is either the `index.js` file if it exists or the `Client.js`.